airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -473,7 +476,69 @@ def request_params(
         return params
 
 
-class SourceZendeskIncrementalExportStream(SourceZendeskSupportCursorPaginationStream):
+class SourceZendeskSupportCursorPaginationStream(SourceZendeskSupportFullRefreshStream):
+    """
+    Only used by streams modified by rudderstack
+    Implements cursor pagination as most streams support cursor pagination.
+    Where it is not supported such streams will override some of this functionality to
+    use offset pagination.
+    """
+
+    cursor_field = "updated_at"
+    next_page_field = "next_page"
+    prev_start_time = None
+
+    @property
+    def state_checkpoint_interval(self) -> Optional[int]:
+        """
+        Will allow the connector send state messages more frequent and not only at the end of the sync.
+        """
+        return 1000
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        # try to save maximum value of a cursor field
+        old_value = str((current_stream_state or {}).get(self.cursor_field, ""))
+        new_value = str((latest_record or {}).get(self.cursor_field, ""))
+        return {self.cursor_field: max(new_value, old_value)}
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if self._ignore_pagination:
+            return None
+        """
+        https://developer.zendesk.com/documentation/api-basics/pagination/paginating-through-lists-using-cursor-pagination/#when-to-stop-paginating
+        """
+        meta = response.json().get("meta", {})
+        return {"page[after]": meta.get("after_cursor")} if meta.get("has_more") else None
+
+    def check_stream_state(self, stream_state: Mapping[str, Any] = None):
+        """
+        Returns the state value, if exists. Otherwise, returns user defined `Start Date`.
+        """
+        state = stream_state.get(self.cursor_field) or self._start_date if stream_state else self._start_date
+        return calendar.timegm(pendulum.parse(state).utctimetuple())
```

Comment:
Results from printing request params of triggers sync run:
{'start_time': 1625598362, 'page[size]': 100, 'sort_order': 'asc'} - 1st call
{'start_time': 1625598362, 'page[size]': 100, 'sort_order': 'asc', 'page[after]': 'token'} - subsequent calls

I don't see `start_time` param in api documentation of triggers. Manually when I make a request after adding `start_time` param it is not filtering records. But when I run a **sync it is working as intended**. Maybe I am missing something in code. Will check. But we **can be confident about connector functionality.**

Reply:
Got it. Also from @am6010 's [comment](https://github.com/rudderlabs/airbyte/pull/138#discussion_r1268308160) we only need to make sure `updated_at` state we save gets used correctly in first API call of a sync as in it starts fetching records from that time.

Reply:
Start_time seems to be not supported for all resources.

From Airbyte [doc](https://docs.airbyte.com/integrations/sources/zendesk-support/):
```
Incremental (standard server-side, where API returns only the data updated or generated since the last sync)
Client-Side Incremental (API returns all available data and connector filters out only new records) :::
```

Blendo code too have `skipBeforeTimestamp`
```
    val count = ZendeskParsing.Page(res).count.getOrElse(throw new IllegalArgumentException)
    ZendeskParsing.Page(res).timeIndex.foreach(_indexInTime = _)
 ```

Reply:
Airbyte is fetching all data and filtering on it's end like semi incremental sync. https://github.com/rudderlabs/airbyte/blob/440362aa806a5af1aaf9ba84f9d996b454d9341c/airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py#L198
Makes sense now!

Reply:
Streams supporting incremental natively from API using say start_time param, check if this filtering is still correct. As ideally it should not require this filtering.
Streams not supporting incremental natively from API, as in client side filtering. Just double check.
For us we can add a top level comment for all streams behavior as well in streams.py.